parlai/agents/rag/retrievers.py

```diff
@@ -640,6 +640,13 @@ def index_retrieve(
             vectors = vectors[:, :, :-1]
         # recompute exact FAISS scores
         scores = torch.bmm(query.unsqueeze(1), vectors.transpose(1, 2)).squeeze(1)
+        if torch.isnan(scores).sum().item():
+            logging.error(
+                '\n[ Document scores are NaN; please look into the built index. ]\n'
+                '[ If using a compressed index, try building an exact index: ]\n'
+                '[ $ python index_dense_embeddings --indexer-type exact... ]'
+            )
```

Comment:
Should the error message also mention something about replacing some dummy numbers for score?

Reply:
don't think it's necessary